### PR TITLE
feat: auth rework

### PR DESCRIPTION
**Pull request by BrickheadJohnny**
Opening this PR on the `rewards` branch to avoid strange conflicts.

### CODE REVIEW

**Review by dominik-stumpf — APPROVED** [review]
Nice work! Why not point this at `add-guild-page`?

**Comment by BrickheadJohnny** [comment]
> Nice work! Why not point this at `add-guild-page`?

We're fetching some entities differently on the `rewards` branch & I opened this branch from that one, so the diff is crazy if I point this PR to the `add-guild-page` branch. We actually need the auth for the Discord reward (which we'll implement on the `rewards` branch), so I thought it's okay to point this PR to that one. 😉

**Comment by dominik-stumpf** [comment]
> > Nice work! Why not point this at `add-guild-page`?
> 
> We're fetching some entities differently on the `rewards` branch & I opened this branch from that one, so the diff is crazy if I point this PR to the `add-guild-page` branch. We actually need the auth for the Discord reward (which we'll implement on the `rewards` branch), so I thought it's okay to point this PR to that one. 😉

Okay, I will be sure to wait for `reward` until the error handling is merged.

**Pull request merged**
